core/commands/ping.go

```diff
@@ -0,0 +1,148 @@
+package commands
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"time"
+
+	cmds "github.com/jbenet/go-ipfs/commands"
+	peer "github.com/jbenet/go-ipfs/p2p/peer"
+	u "github.com/jbenet/go-ipfs/util"
+
+	context "github.com/jbenet/go-ipfs/Godeps/_workspace/src/code.google.com/p/go.net/context"
+)
+
+const kPingTimeout = 10 * time.Second
+
+type PingResult struct {
+	Success bool
+	Time    time.Duration
+	Text    string
+}
+
+var PingCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "send echo request packets to IPFS hosts",
+		Synopsis: `
+ipfs ping <peer.ID> - Send pings to a peer using the routing system to discover its address
+`,
+		ShortDescription: `
+ipfs ping is a tool to find a node (in the routing system),
+send pings, wait for pongs, and print out round-trip latency information.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("peer ID", true, true, "ID of peer to be pinged"),
+	},
+	Options: []cmds.Option{
+		cmds.IntOption("count", "n"),
```

Comment:
Ah, the function signature just said `names ...string` or something. Was misleading.